qt-models/divelocationmodel.cpp

```diff
@@ -224,10 +224,12 @@ DiveSiteSortedModel::DiveSiteSortedModel()
 QStringList DiveSiteSortedModel::allSiteNames() const
 {
 	QStringList locationNames;
-	int num = rowCount();
-	for (int i = 0; i < num; i++) {
-		int idx = mapToSource(index(i, 0)).row();
-		locationNames << QString(dive_site_table.dive_sites[idx]->name);
+	if (dive_site_table.dive_sites) {
```

Comment:
This one here looks wrong. It still can crash if rowCount and nr don't match. You want to loop over the correct number. But it would be important to find out why model and core are out if sync.

Reply:
I agree with both points. Having said that. if you look at the alternative stack trace in #2221 it looked like we were simply dereferencing a NULL pointer... 
The new version hasn't been out long enough to confidently claim that this fixed #2221, but at least I haven't seen it recur with the new version.

Reply:
Looking at this again - this must be bogus.

`dive_site_table` is a global object. If for some reason `dive_site_table.nr` is > 0 and `dive_site_table.dive_sites` is null, the core structures are corrupted. The application will crash on the next attempt to access a dive sites. If not here - elsewhere.

What could happen though is that the model was not updated properly and that `mapToSource` returns an invalid index. Therefore I suggest adding a
```
		if (idx < 0 || idx >= dive_site_table.nr) {
			log_some_warning_visible_on_mobile();
			continue;
		}

Reply:
E.g. `setupDivesite()` in qmlmanager.c does not appear to reset the DiveSiteSortedModel - which will lead to invalid indexes on mapping and could explain that crash. That should be reproducible.